components/ws-manager-bridge/src/bridge.ts

```diff
@@ -504,9 +504,27 @@ export class WorkspaceManagerBridge implements Disposable {
             const actuallyRunningInstances = await client.getWorkspaces(ctx, new GetWorkspacesRequest());
             actuallyRunningInstances.getStatusList().forEach((s) => runningInstancesIdx.delete(s.getId()));
 
+            // runningInstancesIdx only contains instances that ws-manager is not aware of
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
-                if (instance.status.phase !== "running") {
+                const phase = instance.status.phase;
+                if (phase !== "running") {
+                    // This below if block is to validate the planned fix
+                    if (
+                        phase === "pending" ||
+                        phase === "creating" ||
+                        phase === "initializing" ||
+                        (phase === "stopping" &&
+                            instance.stoppingTime &&
+                            durationLongerThanSeconds(Date.parse(instance.stoppingTime), 10))
```

Comment:
@laushinka One more comment on the timeout discussion (sorry for doing this out of band, but couldn't wrap my head around yesterday): We have to make sure that short disconnects between application and workspace clusters don't lead to falsely stopped workspaces. We've seen instances in the past where we had problems on and off for a couple of **minutes** (might have been due to bad gRPC connection options, might or might not be better by now). With this change, this would lead to a situation where we mark the workspace as stopped in the database, while users can use and work in their workspaces perfectly fine. :grimacing: 

I guess we just be very carefule with this change. Let's monitor this for now, and a couple of days. Plus double-check if we find hints in the logs on bridge -> ws-manager re-connects (this should be the [relevant line](https://github.com/gitpod-io/gitpod/blob/2823606369b667188187b1e8ee9a75cbe968633b/components/ws-manager-bridge/src/wsman-subscriber.ts#L38) in code).